test/lifecycle_test.dart

```diff
@@ -448,7 +449,7 @@ void sharedLifecycleTests<T extends react.Component>({
 }) {
   group('(shared behavior)', () {
     group('default props', () {
-      test('getDefaultProps() is only called once per component factory and cached', () {
+      test('defaultProps is only called once per component factory and cached', () {
```

Comment:
We typically leave test coverage around for deprecated stuff - I'd like to see the `getDefaultProps` test moved somewhere that it can still be run for `Component` only.

Reply:
The test for both component2 defaultProps and component1 getDefaultProps is still there maybe we could instead just do: 
```suggestion
      test('is only called once per component factory and cached', () {
```